src/osd/scrubber_common.h

```diff
@@ -12,6 +12,24 @@ namespace ceph {
 class Formatter;
 }
 
+struct PGPool;
+
+namespace Scrub {
+  class ReplicaReservations;
+}
+
+/// Facilitating scrub-realated object access to private PG data
```

Comment:
Is this actually useful?  Seems like you could simply declare *Listener to be a friend of PG without much loss of expressiveness.

Reply:
Not sure if I've answered that one:
the ScrubberPasskey is an existing mechanism, used by the rest of the scrub-related objects.
As that number of those objects increased, the PG became too "friendly".

Note that all I did here is to move the Passkey to another '.h' file

Reply:
As a reader/reviewer, it's just another type I have to lookup every time I see commits changing this portion of the code base.  The two interfaces you've defined already sufficiently clarify what portions of PG the scrubber objects can access, and this really just seems like extra cognitive overhead.

Reply:
Let's discuss this online. The passkey goal was "PG-oriented" - limiting the amount of code that
can potentially modify PG class internals (a different type of cognitive load). Not really related to
scrubber's code design.
Now that it's there - I don't think this PR is the right place to undo it.

Reply:
Ok